File: suc/kotlin/008.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.BottomAppBar
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.NavigationBar
import androidx.compose.material3.NavigationBarItem
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "011_HealthDashboard_en"
private const val UI_TYPE = "Health Fitness Dashboard"
private const val STYLE_THEME = "Vibrant Gradient"
private const val LANG = "en"
private const val BASELINE_SIZE = "1280x720"

object AppTokens {
    object Colors {
        val primary = Color(0xFF6366F1)
        val secondary = Color(0xFFEC4899)
        val tertiary = Color(0xFF10B981)
        val background = Color(0xFFF8FAFC)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFF1F5F9)
        val outline = Color(0xFFE2E8F0)
        val success = Color(0xFF10B981)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF0F172A)
        val onSurface = Color(0xFF0F172A)
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.Bold, lineHeight = 30.sp, letterSpacing = (-0.25).sp)
        val headline = TextStyle(fontSize = 20.sp, fontWeight = FontWeight.SemiBold, lineHeight = 26.sp, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Medium, lineHeight = 22.sp, letterSpacing = 0.sp)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal, lineHeight = 20.sp, letterSpacing = 0.sp)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium, lineHeight = 16.sp, letterSpacing = 0.sp)
        val small = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Normal, lineHeight = 14.sp, letterSpacing = 0.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(8.dp)
        val medium = RoundedCornerShape(12.dp)
        val large = RoundedCornerShape(16.dp)
    }

    object Spacing {
        val xs = 2.dp
        val sm = 4.dp
        val md = 8.dp
        val lg = 12.dp
        val xl = 16.dp
        val xxl = 20.dp
        val xxxl = 24.dp
    }

    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 4.dp, 1.dp, 0.15f)
        val level2 = ShadowSpec(3.dp, 8.dp, 2.dp, 0.18f)
        val level3 = ShadowSpec(6.dp, 12.dp, 3.dp, 0.22f)
        val level4 = ShadowSpec(8.dp, 16.dp, 4.dp, 0.25f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            extraSmall = AppTokens.Shapes.small,
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large,
            extraLarge = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class HealthMetric(val id: Int, val title: String, val value: String, val unit: String, val progress: Float, val color: Color)
data class Workout(val id: Int, val name: String, val duration: String, val calories: String, val iconColor: Color)
data class NavItem(val id: Int, val label: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val selectedNav = remember { mutableStateOf(0) }
    val metrics = remember {
        listOf(
            HealthMetric(1, "Heart", "72", "bpm", 0.6f, AppTokens.Colors.secondary),
            HealthMetric(2, "Steps", "842", "steps", 0.8f, AppTokens.Colors.primary),
            HealthMetric(3, "Sleep", "7.2", "hours", 0.9f, AppTokens.Colors.tertiary),
            HealthMetric(4, "Calories", "420", "kcal", 0.4f, AppTokens.Colors.error)
        )
    }
    val workouts = remember {
        listOf(
            Workout(1, "Morning Run", "32 min", "284 kcal", AppTokens.Colors.primary),
            Workout(2, "Yoga Flow", "45 min", "156 kcal", AppTokens.Colors.secondary),
            Workout(3, "Strength Training", "28 min", "320 kcal", AppTokens.Colors.tertiary)
        )
    }
    val navItems = remember {
        listOf(
            NavItem(0, "Dashboard"),
            NavItem(1, "Workouts"),
            NavItem(2, "Nutrition"),
            NavItem(3, "Profile")
        )
    }

    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
                    containerColor = Color.Transparent
                ),
                title = {
                    Text(
                        text = "Health Dashboard",
                        style = MaterialTheme.typography.displayLarge,
                        color = MaterialTheme.colorScheme.onBackground
                    )
                }
            )
        },
        bottomBar = {
            NavigationBar(
                containerColor = MaterialTheme.colorScheme.surface,
                tonalElevation = AppTokens.ElevationMapping.level2.elevation
            ) {
                navItems.forEach { item ->
                    NavigationBarItem(
                        selected = selectedNav.value == item.id,
                        onClick = { selectedNav.value = item.id },
                        icon = {
                            Box(
                                modifier = Modifier
                                    .size(20.dp)
                                    .background(
                                        if (selectedNav.value == item.id) MaterialTheme.colorScheme.primary
                                        else MaterialTheme.colorScheme.outline,
                                        CircleShape
                                    )
                            )
                        },
                        label = {
                            Text(
                                text = item.label,
                                style = MaterialTheme.typography.labelMedium
                            )
                        }
                    )
                }
            }
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { padding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(horizontal = AppTokens.Spacing.md),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg),
            contentPadding = PaddingValues(bottom = AppTokens.Spacing.xxxl)
        ) {
            item {
                Card(
                    shape = AppTokens.Shapes.large,
                    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primary),
                    elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level3.elevation),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Column(
                        modifier = Modifier.padding(AppTokens.Spacing.xl),
                        verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                    ) {
                        Text(
                            text = "Good Morning!",
                            style = MaterialTheme.typography.titleMedium,
                            color = MaterialTheme.colorScheme.onPrimary
                        )
                        Text(
                            text = "Ready for your workout?",
                            style = MaterialTheme.typography.headlineMedium,
                            color = MaterialTheme.colorScheme.onPrimary
                        )
                        Button(
                            onClick = {},
                            colors = ButtonDefaults.buttonColors(
                                containerColor = MaterialTheme.colorScheme.onPrimary,
                                contentColor = MaterialTheme.colorScheme.primary
                            ),
                            shape = AppTokens.Shapes.medium,
                            modifier = Modifier
                                .fillMaxWidth()
                                .height(44.dp)
                        ) {
                            Text(
                                text = "Start Workout",
                                style = MaterialTheme.typography.titleMedium
                            )
                        }
                    }
                }
            }

            item {
                Text(
                    text = "Today's Metrics",
                    style = MaterialTheme.typography.headlineMedium,
                    color = MaterialTheme.colorScheme.onBackground,
                    modifier = Modifier.padding(horizontal = AppTokens.Spacing.sm)
                )
            }

            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
                ) {
                    metrics.forEach { metric ->
                        Card(
                            shape = AppTokens.Shapes.medium,
                            colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                            elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation),
                            modifier = Modifier.weight(1f)
                        ) {
                            Column(
                                modifier = Modifier.padding(AppTokens.Spacing.md),
                                horizontalAlignment = Alignment.CenterHorizontally,
                                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)
                            ) {
                                Box(
                                    modifier = Modifier
                                        .size(36.dp)
                                        .background(metric.color, CircleShape)
                                )
                                Text(
                                    text = metric.value,
                                    style = MaterialTheme.typography.headlineMedium,
                                    color = MaterialTheme.colorScheme.onSurface
                                )
                                Text(
                                    text = metric.title,
                                    style = MaterialTheme.typography.labelMedium,
                                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                                )
                            }
                        }
                    }
                }
            }

            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = "Recent Workouts",
                        style = MaterialTheme.typography.headlineMedium,
                        color = MaterialTheme.colorScheme.onBackground
                    )
                    Button(
                        onClick = {},
                        colors = ButtonDefaults.buttonColors(
                            containerColor = Color.Transparent,
                            contentColor = MaterialTheme.colorScheme.primary
                        ),
                        elevation = ButtonDefaults.buttonElevation(0.dp)
                    ) {
                        Text(
                            text = "View All",
                            style = MaterialTheme.typography.labelMedium
                        )
                    }
                }
            }

            items(workouts) { workout ->
                Card(
                    shape = AppTokens.Shapes.medium,
                    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                    elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Row(
                        modifier = Modifier.padding(AppTokens.Spacing.md),
                        horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Box(
                            modifier = Modifier
                                .size(44.dp)
                                .background(workout.iconColor, AppTokens.Shapes.medium)
                        )
                        Column(
                            modifier = Modifier.weight(1f),
                            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)
                        ) {
                            Text(
                                text = workout.name,
                                style = MaterialTheme.typography.titleMedium,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                            Row(
                                horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
                            ) {
                                Text(
                                    text = workout.duration,
                                    style = MaterialTheme.typography.bodyMedium,
                                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                                )
                                Text(
                                    text = workout.calories,
                                    style = MaterialTheme.typography.bodyMedium,
                                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                                )
                            }
                        }
                        Button(
                            onClick = {},
                            colors = ButtonDefaults.buttonColors(
                                containerColor = MaterialTheme.colorScheme.primary,
                                contentColor = MaterialTheme.colorScheme.onPrimary
                            ),
                            shape = AppTokens.Shapes.small,
                            modifier = Modifier
                                .height(32.dp)
                                .width(70.dp)
                        ) {
                            Text(
                                text = "Start",
                                style = AppTokens.TypographyTokens.small
                            )
                        }
                    }
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.navigationBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.navigationBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFFFFFFF, widthDp = 360, heightDp = 640)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/008.swift
import SwiftUI

enum HealthDashboardTokens {
    enum Colors {
        static let primary = Color(red: 99 / 255, green: 102 / 255, blue: 241 / 255)
        static let secondary = Color(red: 236 / 255, green: 72 / 255, blue: 153 / 255)
        static let tertiary = Color(red: 16 / 255, green: 185 / 255, blue: 129 / 255)
        static let background = Color(red: 248 / 255, green: 250 / 255, blue: 252 / 255)
        static let surface = Color.white
        static let surfaceVariant = Color(red: 241 / 255, green: 245 / 255, blue: 249 / 255)
        static let outline = Color(red: 226 / 255, green: 232 / 255, blue: 240 / 255)
        static let error = Color(red: 239 / 255, green: 68 / 255, blue: 68 / 255)
        static let onPrimary = Color.white
        static let onBackground = Color(red: 15 / 255, green: 23 / 255, blue: 42 / 255)
        static let onSurface = Color(red: 15 / 255, green: 23 / 255, blue: 42 / 255)
    }

    enum Typography {
        static let display = Font.system(size: 24, weight: .bold)
        static let headline = Font.system(size: 20, weight: .semibold)
        static let title = Font.system(size: 16, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
        static let small = Font.system(size: 10, weight: .regular)
    }

    enum Shapes {
        static let small: CGFloat = 8
        static let medium: CGFloat = 12
        static let large: CGFloat = 16
    }

    enum Spacing {
        static let xs: CGFloat = 2
        static let sm: CGFloat = 4
        static let md: CGFloat = 8
        static let lg: CGFloat = 12
        static let xl: CGFloat = 16
        static let xxl: CGFloat = 20
        static let xxxl: CGFloat = 24
    }

    struct ShadowSpec {
        let radius: CGFloat
        let y: CGFloat
        let opacity: Double
    }

    enum Elevation {
        static let level1 = ShadowSpec(radius: 4, y: 1, opacity: 0.15)
        static let level2 = ShadowSpec(radius: 8, y: 2, opacity: 0.18)
        static let level3 = ShadowSpec(radius: 12, y: 3, opacity: 0.22)
        static let level4 = ShadowSpec(radius: 16, y: 4, opacity: 0.25)
    }
}

struct HealthMetric: Identifiable {
    let id: Int
    let title: String
    let value: String
    let unit: String
    let color: Color
}

struct Workout: Identifiable {
    let id: Int
    let name: String
    let duration: String
    let calories: String
    let iconColor: Color
}

struct NavItem: Identifiable {
    let id: Int
    let label: String
}

struct HealthDashboardAppView: View {
    @State private var selectedNav: Int = 0

    private let metrics: [HealthMetric] = [
        HealthMetric(id: 1, title: "Heart", value: "72", unit: "bpm", color: HealthDashboardTokens.Colors.secondary),
        HealthMetric(id: 2, title: "Steps", value: "842", unit: "steps", color: HealthDashboardTokens.Colors.primary),
        HealthMetric(id: 3, title: "Sleep", value: "7.2", unit: "hours", color: HealthDashboardTokens.Colors.tertiary),
        HealthMetric(id: 4, title: "Calories", value: "420", unit: "kcal", color: HealthDashboardTokens.Colors.error)
    ]

    private let workouts: [Workout] = [
        Workout(id: 1, name: "Morning Run", duration: "32 min", calories: "284 kcal", iconColor: HealthDashboardTokens.Colors.primary),
        Workout(id: 2, name: "Yoga Flow", duration: "45 min", calories: "156 kcal", iconColor: HealthDashboardTokens.Colors.secondary),
        Workout(id: 3, name: "Strength Training", duration: "28 min", calories: "320 kcal", iconColor: HealthDashboardTokens.Colors.tertiary)
    ]

    private let navItems: [NavItem] = [
        NavItem(id: 0, label: "Dashboard"),
        NavItem(id: 1, label: "Workouts"),
        NavItem(id: 2, label: "Nutrition"),
        NavItem(id: 3, label: "Profile")
    ]

    private let bottomBarHeight: CGFloat = 72

    var body: some View {
        ZStack(alignment: .bottom) {
            HealthDashboardTokens.Colors.background
                .ignoresSafeArea()

            VStack(spacing: 0) {
                topBar
                    .padding(.top, 40)
                    .padding(.bottom, 20)
                ScrollView(showsIndicators: false) {
                    LazyVStack(spacing: HealthDashboardTokens.Spacing.lg) {
                        heroCard
                        sectionTitle("Today's Metrics")
                        metricsRow
                        recentWorkoutsHeader
                            .padding(.vertical, 10)
                        ForEach(workouts) { workout in
                            workoutCard(workout)
                        }
                    }
                    .padding(.horizontal, HealthDashboardTokens.Spacing.md)
                    .padding(.bottom, bottomBarHeight + HealthDashboardTokens.Spacing.lg)
                }
            }

            bottomNavigation
                .padding(.horizontal, HealthDashboardTokens.Spacing.md)
                .padding(.bottom, HealthDashboardTokens.Spacing.md)
        }
        .statusBarHidden(true)
    }

    private var topBar: some View {
        Text("Health Dashboard")
            .font(HealthDashboardTokens.Typography.display)
            .foregroundColor(HealthDashboardTokens.Colors.onBackground)
            .frame(maxWidth: .infinity)
            .padding(.top, HealthDashboardTokens.Spacing.xxl)
            .padding(.bottom, HealthDashboardTokens.Spacing.md)
    }

    private var heroCard: some View {
        VStack(alignment: .leading, spacing: HealthDashboardTokens.Spacing.sm) {
            Text("Good Morning!")
                .font(HealthDashboardTokens.Typography.title)
                .foregroundColor(HealthDashboardTokens.Colors.onPrimary)
            Text("Ready for your workout?")
                .font(HealthDashboardTokens.Typography.headline)
                .foregroundColor(HealthDashboardTokens.Colors.onPrimary)
            Button(action: {}) {
                Text("Start Workout")
                    .font(HealthDashboardTokens.Typography.title)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(height: 44)
            .background(
                RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.medium, style: .continuous)
                    .fill(HealthDashboardTokens.Colors.onPrimary)
            )
            .foregroundColor(HealthDashboardTokens.Colors.primary)
        }
        .padding(HealthDashboardTokens.Spacing.xl)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.large, style: .continuous)
                .fill(HealthDashboardTokens.Colors.primary)
        )
        .shadow(color: Color.black.opacity(HealthDashboardTokens.Elevation.level3.opacity),
                radius: HealthDashboardTokens.Elevation.level3.radius,
                x: 0,
                y: HealthDashboardTokens.Elevation.level3.y)
    }

    private func sectionTitle(_ title: String) -> some View {
        Text(title)
            .font(.system(size: 14))
            .foregroundColor(HealthDashboardTokens.Colors.onBackground)
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, HealthDashboardTokens.Spacing.sm)
    }

    private var metricsRow: some View {
        HStack(spacing: HealthDashboardTokens.Spacing.md) {
            ForEach(metrics) { metric in
                HStack(){
                    VStack(spacing: HealthDashboardTokens.Spacing.xs) {
                        Circle()
                            .fill(metric.color)
                            .frame(width: 36, height: 36)
                        Text(metric.value)
                            .font(HealthDashboardTokens.Typography.headline)
                            .foregroundColor(HealthDashboardTokens.Colors.onSurface)
                        Text(metric.title)
                            .font(HealthDashboardTokens.Typography.label)
                            .foregroundColor(HealthDashboardTokens.Colors.onSurface.opacity(0.7))
                    }
                    Spacer()
                }
                .padding(HealthDashboardTokens.Spacing.md)
                .frame(alignment: .leading)
                .background(
                    RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.medium, style: .continuous)
                        .fill(HealthDashboardTokens.Colors.surface)
                )
                .shadow(color: Color.black.opacity(HealthDashboardTokens.Elevation.level2.opacity),
                        radius: HealthDashboardTokens.Elevation.level2.radius,
                        x: 0,
                        y: HealthDashboardTokens.Elevation.level2.y)
            }
        }
    }

    private var recentWorkoutsHeader: some View {
        HStack {
            Text("Recent Workouts")
                .font(HealthDashboardTokens.Typography.headline)
                .foregroundColor(HealthDashboardTokens.Colors.onBackground)
            Spacer()
            Button(action: {}) {
                Text("View All")
                    .font(HealthDashboardTokens.Typography.label)
                    .padding(.trailing, 20)
            }
            .buttonStyle(.plain)
            .foregroundColor(HealthDashboardTokens.Colors.primary)
        }
    }

    private func workoutCard(_ workout: Workout) -> some View {
        HStack(spacing: HealthDashboardTokens.Spacing.md) {
            RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.medium, style: .continuous)
                .fill(workout.iconColor)
                .frame(width: 44, height: 44)

            VStack(alignment: .leading, spacing: HealthDashboardTokens.Spacing.xs) {
                Text(workout.name)
                    .font(HealthDashboardTokens.Typography.title)
                    .foregroundColor(HealthDashboardTokens.Colors.onSurface)
                HStack(spacing: HealthDashboardTokens.Spacing.lg) {
                    Text(workout.duration)
                        .font(HealthDashboardTokens.Typography.body)
                        .foregroundColor(HealthDashboardTokens.Colors.onSurface.opacity(0.7))
                    Text(workout.calories)
                        .font(HealthDashboardTokens.Typography.body)
                        .foregroundColor(HealthDashboardTokens.Colors.onSurface.opacity(0.7))
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            Button(action: {}) {
                Text("Start")
                    .font(HealthDashboardTokens.Typography.small)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(width: 70, height: 32)
            .background(
                RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.small, style: .continuous)
                    .fill(HealthDashboardTokens.Colors.primary)
            )
            .foregroundColor(HealthDashboardTokens.Colors.onPrimary)
        }
        .padding(HealthDashboardTokens.Spacing.lg)
        .background(
            RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.medium, style: .continuous)
                .fill(HealthDashboardTokens.Colors.surface)
        )
        .shadow(color: Color.black.opacity(HealthDashboardTokens.Elevation.level1.opacity),
                radius: HealthDashboardTokens.Elevation.level1.radius,
                x: 0,
                y: HealthDashboardTokens.Elevation.level1.y)
    }

    private var bottomNavigation: some View {
        HStack(spacing: HealthDashboardTokens.Spacing.md) {
            ForEach(navItems) { item in
                Button(action: { selectedNav = item.id }) {
                    VStack(spacing: HealthDashboardTokens.Spacing.xs) {
                        Circle()
                            .fill(selectedNav == item.id ? HealthDashboardTokens.Colors.primary : HealthDashboardTokens.Colors.outline)
                            .frame(width: 20, height: 20)
                        Text(item.label)
                            .font(HealthDashboardTokens.Typography.label)
                            .foregroundColor(HealthDashboardTokens.Colors.onSurface)
                    }
                    .frame(maxWidth: .infinity)
                }
                .buttonStyle(.plain)
            }
        }
        .padding(.vertical, HealthDashboardTokens.Spacing.sm)
        .padding(.horizontal, HealthDashboardTokens.Spacing.md)
        .frame(maxWidth: .infinity)
        .background(
            RoundedRectangle(cornerRadius: HealthDashboardTokens.Shapes.large, style: .continuous)
                .fill(HealthDashboardTokens.Colors.surface)
        )
        .shadow(color: Color.black.opacity(HealthDashboardTokens.Elevation.level2.opacity),
                radius: HealthDashboardTokens.Elevation.level2.radius,
                x: 0,
                y: HealthDashboardTokens.Elevation.level2.y)
    }
}

struct HealthDashboardAppView_Previews: PreviewProvider {
    static var previews: some View {
        HealthDashboardAppView()
    }
}

@main
struct HealthDashboardApp: App {
    var body: some Scene {
        WindowGroup {
            HealthDashboardAppView()
                .statusBarHidden(true)
                .background(HealthDashboardTokens.Colors.background.ignoresSafeArea())
        }
    }
}
